source/features/add-co-authored-by.js

```diff
@@ -0,0 +1,183 @@
+import {h} from 'dom-chef';
+import select from 'select-dom';
+import * as api from '../libs/api';
+import {escapeForGql} from '../libs/utils';
+import {getOwnerAndRepo, getPRNumber} from '../libs/page-detect';
+
+const coAuthorData = {};
+
+async function fetchCoAuthoredData() {
+	const {ownerName, repoName} = getOwnerAndRepo();
+	const prNumber = getPRNumber();
+
+	if (!ownerName || !repoName || !prNumber) {
+		return;
+	}
+
+	let apiResponse;
+	try {
+		apiResponse = await api.v4(
+			`{
+				repository(owner: "${ownerName}", name: "${repoName}") {
+					pullRequest(number: ${prNumber}) {
+						commits(first: 100) {
+							nodes {
+								commit {
+									author {
+										email
+										user {
+											databaseId
+											login
+											name
+											email
+										}
+									}
+								}
+							}
+						}
+						reviews(first: 100) {
+							nodes {
+								author {
+									login
+								}
+							}
+						}
+						comments(first: 100) {
+							nodes {
+								author {
+									login
+								}
+							}
+						}
+					}
+				}
+			}`
+		);
+	} catch (error) {
+		disableCoAuthorButton(error);
+		return;
+	}
+	const {data: contributorData, errors} = apiResponse;
+
+	if (errors && errors[0].message.indexOf('the following scopes: [\'user:email\', \'read:user\']') >= 0) {
+		disableCoAuthorButton('To add co-authors, please add the "user:email" to your personal token.');
+		return;
+	}
+
+	coAuthorData.userData = [];
+	coAuthorData.committers = new Set();
+
+	for (const commit of contributorData.repository.pullRequest.commits.nodes) {
```

Comment:
`const {author, user} = commit.commit;`

To make the whole thing a bit more readable